commit f512caac5b6f30fb6c226b73350d0180bdd41d38
Author: Alberto Donato <alberto.donato@canonical.com>
Date:   Thu Sep 21 09:56:12 2017 +0200

    shared/version: include OS, architecture and possibly kernel and distro info in User-Agent
    
    Signed-off-by: Alberto Donato <alberto.donato@canonical.com>

diff --git a/shared/version/flex.go b/shared/version/flex.go
index 5feeedb..48a3537 100644
--- a/shared/version/flex.go
+++ b/shared/version/flex.go
@@ -1,10 +1,33 @@
 package version
 
+import (
+	"fmt"
+	"runtime"
+	"strings"
+
+	"github.com/lxc/lxd/shared/osarch"
+)
+
 // Version contains the LXD version number
 var Version = "2.18"
 
 // UserAgent contains a string suitable as a user-agent
-var UserAgent = "LXD " + Version
+var UserAgent = getUserAgent()
 
 // APIVersion contains the API base version. Only bumped for backward incompatible changes.
 var APIVersion = "1.0"
+
+func getUserAgent() string {
+	archID, err := osarch.ArchitectureId(runtime.GOARCH)
+	if err != nil {
+		panic(err)
+	}
+	arch, err := osarch.ArchitectureName(archID)
+	if err != nil {
+		panic(err)
+	}
+
+	tokens := []string{strings.Title(runtime.GOOS), arch}
+	tokens = append(tokens, getPlatformVersionStrings()...)
+	return fmt.Sprintf("LXD %s (%s)", Version, strings.Join(tokens, "; "))
+}
